### PR TITLE
replace offset args with new GinkgoHelper

### PR DESCRIPTION
**Pull request by huww98**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

it is simpler, and easier to maintain

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/sanity/resources.go [hunk 1]
```diff
@@ -77,88 +77,65 @@ type Resources struct {
...
 	}
 	return conpubvol, err
 }
 
 // registerVolume adds or updates an entry for given volume.
-func (cl *Resources) registerVolume(offset int, id string, info volumeInfo) {
-	ExpectWithOffset(offset, id).NotTo(BeEmpty(), "volume ID is empty")
-	ExpectWithOffset(offset, info).NotTo(BeNil(), "volume info is nil")
+func (cl *Resources) registerVolume(id string, info volumeInfo) {
+	Expect(id).NotTo(BeEmpty(), "volume ID is empty")
```
Comment (pohly) [inline pkg/sanity/resources.go:137]: Missing `ginkgo.GinkgoHelper` here.

File: pkg/sanity/resources.go [hunk 2]
```diff
@@ -171,65 +148,69 @@ func (cl *Resources) registerVolume(offset int, id string, info volumeInfo) {
...
 	cl.mutex.Lock()
 	defer cl.mutex.Unlock()
-	cl.registerSnapshotNoLock(offset+1, id)
+	cl.registerSnapshotNoLock(id)
 }
 
-func (cl *Resources) registerSnapshotNoLock(offset int, id string) {
-	ExpectWithOffset(offset, id).NotTo(BeEmpty(), "ID for register snapshot is missing")
+func (cl *Resources) registerSnapshotNoLock(id string) {
+	Expect(id).NotTo(BeEmpty(), "ID for register snapshot is missing")
```
Comment (pohly) [inline pkg/sanity/resources.go:198]: Missing `ginkgo.GinkgoHelper` here.

File: pkg/sanity/resources.go [hunk 3]
```diff
@@ -171,65 +148,69 @@ func (cl *Resources) registerVolume(offset int, id string, info volumeInfo) {
...
 	cl.mutex.Lock()
 	defer cl.mutex.Unlock()
-	cl.unregisterResourceNoLock(offset+1, id)
+	cl.unregisterResourceNoLock(id)
 }
 
-func (cl *Resources) unregisterResourceNoLock(offset int, id string) {
-	ExpectWithOffset(offset, id).NotTo(BeEmpty(), "ID for unregister resource is missing")
+func (cl *Resources) unregisterResourceNoLock(id string) {
+	Expect(id).NotTo(BeEmpty(), "ID for unregister resource is missing")
```
Comment (pohly) [inline pkg/sanity/resources.go:213]: Missing `ginkgo.GinkgoHelper` here.

**Review by pohly — CHANGES_REQUESTED** [review]
`ginkgo.GinkgoHelper` didn't exist when this code was written. Using it indeed is simpler.

We want to treat all methods as helpers, right? Then some `ginkgo.GinkgoHelper` calls are missing.

**Comment by huww98** [comment]
> We want to treat all methods as helpers, right?

Yes, but so long as one of the callers has `GinkgoHelper()`, it should be fine. And I think `registerXxx()` is not intended to be called outside of resources.go.

**Comment by pohly** [comment]
> Yes, but so long as one of the callers has GinkgoHelper(), it should be fine.

That's not my understanding of how that function works. [GinkgoHelper](https://pkg.go.dev/github.com/onsi/ginkgo/v2#GinkgoHelper) marks "*the function* it's called in as a test helper" (emphasis mine). It does not mark the function and all functions called by it as helper.

**Pull request merged**